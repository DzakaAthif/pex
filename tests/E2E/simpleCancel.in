BUY 0 Avocado 50 100;
CANCEL 0;
SELL 1 Wine 100 110;
CANCEL 1;
BUY 2 Avocado 50 100;
BUY 3 Avocado 50 101;
BUY 4 Avocado 50 102;
SELL 5 Wine 50 100;
SELL 6 Wine 50 101;
SELL 7 Wine 50 102;
CANCEL 2;
CANCEL 3;
CANCEL 4;
CANCEL 5;
CANCEL 6;
CANCEL 7;