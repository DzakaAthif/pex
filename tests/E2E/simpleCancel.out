[PEX] Starting
[PEX] Trading 2 products: Wine Avocado
[PEX] Created FIFO /tmp/pe_exchange_0
[PEX] Created FIFO /tmp/pe_trader_0
[PEX] Starting trader 0 (./trader1)
[PEX] Connected to /tmp/pe_exchange_0
[PEX] Connected to /tmp/pe_trader_0
[PEX] [T0] Parsing command: <BUY 0 Avocado 50 100>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 1; Sell levels: 0
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 0>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 1 Wine 100 110>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 1
[PEX]		SELL 100 @ $110 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 1>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <BUY 2 Avocado 50 100>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 1; Sell levels: 0
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <BUY 3 Avocado 50 101>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 2; Sell levels: 0
[PEX]		BUY 50 @ $101 (1 order)
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <BUY 4 Avocado 50 102>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 3; Sell levels: 0
[PEX]		BUY 50 @ $102 (1 order)
[PEX]		BUY 50 @ $101 (1 order)
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 5 Wine 50 100>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 1
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 3; Sell levels: 0
[PEX]		BUY 50 @ $102 (1 order)
[PEX]		BUY 50 @ $101 (1 order)
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 6 Wine 50 101>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 2
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 3; Sell levels: 0
[PEX]		BUY 50 @ $102 (1 order)
[PEX]		BUY 50 @ $101 (1 order)
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 7 Wine 50 102>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 3; Sell levels: 0
[PEX]		BUY 50 @ $102 (1 order)
[PEX]		BUY 50 @ $101 (1 order)
[PEX]		BUY 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 2>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 2; Sell levels: 0
[PEX]		BUY 50 @ $102 (1 order)
[PEX]		BUY 50 @ $101 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 3>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 1; Sell levels: 0
[PEX]		BUY 50 @ $102 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 4>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 5>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 2
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 6>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 1
[PEX]		SELL 50 @ $102 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <CANCEL 7>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 0
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] Trader 0 disconnected
[PEX] Trading completed
[PEX] Exchange fees collected: $0
